Src/WindowsServer/WindowsServer.Shared/AzureRoleEnvironmentTelemetryInitializer.cs

```diff
@@ -30,22 +48,41 @@ public AzureRoleEnvironmentTelemetryInitializer()
         public void Initialize(ITelemetry telemetry)
         {
             if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleName))
-            {
-                var name = LazyInitializer.EnsureInitialized(ref this.roleName, AzureRoleEnvironmentContextReader.Instance.GetRoleName);
-                telemetry.Context.Cloud.RoleName = name;
+            {                
+                telemetry.Context.Cloud.RoleName = this.roleName;
             }
 
             if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
-            {
-                var name = LazyInitializer.EnsureInitialized(ref this.roleInstanceName, AzureRoleEnvironmentContextReader.Instance.GetRoleInstanceName);
-                telemetry.Context.Cloud.RoleInstance = name;
+            {                
+                telemetry.Context.Cloud.RoleInstance = this.roleInstanceName;
             }
 
             if (string.IsNullOrEmpty(telemetry.Context.GetInternalContext().NodeName))
+            {                
+                telemetry.Context.GetInternalContext().NodeName = this.roleInstanceName;
+            }
+        }
+
+        /// <summary>
+        /// Searches for the environment variable specific to Azure web applications and confirms if the current application is a web application or not.
+        /// </summary>
+        /// <returns>Boolean, which is true if the current application is an Azure web application.</returns>
+        private static bool IsAppRunningInAzureWebApp()
+        {
+            if (!isAzureWebApp.HasValue)
             {
-                var name = LazyInitializer.EnsureInitialized(ref this.roleInstanceName, AzureRoleEnvironmentContextReader.Instance.GetRoleInstanceName);
-                telemetry.Context.GetInternalContext().NodeName = name;
+                try
+                {
+                    isAzureWebApp = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(WebSiteEnvironmentVariable));
+                }
+                catch (Exception ex)
+                {
+                    WindowsServerEventSource.Log.AccessingEnvironmentVariableFailedWarning(WebSiteEnvironmentVariable, ex.ToString());
```

Comment:
I think we should switch to PlatformImplementation.AccessingEnvironmentVariableFailedWarning and make it public.